core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -700,7 +700,7 @@ public final void post(BuildListener listener) throws Exception {
                 HashSet<String> r = new HashSet<>();
```

Comment:
Should we consider changing this to a sorted set to maintain the original behavior of sorting the culprits? Or is that not relevant in this use case? Not sure offhand.

```java
SortedSet<String> r = new TreeSet<>();
```

Reply:
ah yes good point. not sure we really need it here but best to keep current behaviour